modules/integration/src/test/java/io/ballerina/messaging/broker/integration/standalone/jms/QueueLocalTransactionRollbackTest.java

```diff
@@ -0,0 +1,260 @@
+import javax.jms.Session;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+/**
+ * Test cases written to verify queue consumer and producer behavior in local transaction rollback operation
+ */
+public class QueueLocalTransactionRollbackTest {
+
+    @Parameters({"broker-port"})
+    @Test
+    public void testConsumerRollbackTransaction(String port) throws NamingException, JMSException {
+        String queueName = "testConsumerRollbackTransaction";
+        InitialContext initialContextForQueue = ClientHelper
+                .getInitialContextBuilder("admin", "admin", "localhost", port)
+                .withQueue(queueName)
+                .build();
+
+        ConnectionFactory connectionFactory
+                = (ConnectionFactory) initialContextForQueue.lookup(ClientHelper.CONNECTION_FACTORY);
+        Connection connection = connectionFactory.createConnection();
+        connection.start();
+
+        // send 100 messages
+        Session producerSession = connection.createSession(true, Session.SESSION_TRANSACTED);
+        Queue queue = producerSession.createQueue(queueName);
+        MessageProducer producer = producerSession.createProducer(queue);
+
+        int numberOfMessages = 100;
+        for (int i = 0; i < numberOfMessages; i++) {
+            producer.send(producerSession.createTextMessage("Test message " + i));
+        }
+        // commit all sent messages
+        producerSession.commit();
+        producerSession.close();
+
+        // consume messages
+        Session subscriberSession = connection.createSession(true, Session.SESSION_TRANSACTED);
+        Destination subscriberDestination = (Destination) initialContextForQueue.lookup(queueName);
+        MessageConsumer consumer = subscriberSession.createConsumer(subscriberDestination);
+
+        for (int i = 0; i < numberOfMessages; i++) {
+            Message message = consumer.receive(1000);
+            Assert.assertNotNull(message, "Message #" + i + " was not received");
+        }
+        // rollback all received messages
+        subscriberSession.rollback();
+
+        // consumer receive all messages after rollback
+        int numberOfMessagesAfterRollback = 0;
+        for (int i = 0; i < numberOfMessages; i++) {
+            Message message = consumer.receive(1000);
+            Assert.assertNotNull(message, "Message #" + i + " was not received after rollback");
+            numberOfMessagesAfterRollback++;
+        }
+        Assert.assertEquals(numberOfMessages, numberOfMessagesAfterRollback, "Only "
+                + numberOfMessagesAfterRollback + " messages received after rollback but expect "
+                + numberOfMessages + " messages");
+
+        subscriberSession.commit();
+
+        subscriberSession.close();
+        connection.close();
+    }
+
+    @Parameters({"broker-port"})
+    @Test
+    public void testProducerRollbackTransaction(String port) throws NamingException, JMSException {
```

Comment:
At each test method shall we add a description what is tested at the method?

Reply:
I think method name implies what exactly does in the test case. Also, I have added inline comment some points to elaborate the logic. All test we have written in a similar way. Please let me know if you find any method which couldn't understand by reading the name. Then I'll change it accordingly.